docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+
+```shell
+for i in 0 1; do
+  kubectl exec web-$i -- sh -c 'echo $(hostname) > /usr/share/nginx/html/index.html';
+done
+```
+
+Now delete all pods in the petset:
+
+```shell
+$ kubectl delete po -l app=nginx
+pod "web-0" deleted
+pod "web-1" deleted
+```
+
+Wait for them to come back up, and try to retrieve the previously written hostname through the DNS name of the peer (remember the Pet Set also gives us stable storage, and that the hostname is linked to the DNS name).
+
+```shell
+$ kubectl exec -it web-1 -- curl web-0.nginx
+web-0
+$ kubectl exec -it web-0 -- curl web-1.nginx
+web-1
+```
+
+#### Discovery
+
+A pet can piece together its own identity:
+
+1. Use the [downward api](http://kubernetes.io/docs/user-guide/downward-api/) to find its pod name
+2. Run `hostname` to find its DNS name
+3. Run `mount` or `df` to find its volumes (usually this is unnecessary)
+
+It's not necessary to "discover" the governing Service of a Pet Set, since it's known at creation time (i.e you can pass it down through an environment variable).
+
+Usually Pets in a Pet Set also need to be able to find their peers. In the previous nginx example, we just used `kubectl` to get the names of existing pods, and as humans, we could tell which ones belonged to a given Pet Set. Another way to find peers is by contacting the API server, just like `kubectl`, but that has several disadvantages:
+
+* It makes any startup script non-portable, users need to write specialized code for Kubernetes
+* It introduces several points of failure
+    - the secrets to talk to the API server might not be mounted in the pod
+    - the apiserver Service might be in flux
+    - kube-proxy on the node might not have updated the apiserver endpoint rules etc
+* It introduces an upgrade related failure scenario, since you're deserializing raw json from the apiserver
+* It puts load on the apiserver
+
+Pet Set gives you a way to disover your peers using DNS records. To illustrate this we can use the previous example (note: one usually doesn't `apt-get` in a container).
+
+```shell
+$ kubectl exec -it web-0 /bin/sh
+# apt-get update && apt-get install -y dnsutils
+...
+
+# nslookup -type=srv nginx.default
+Server:		10.0.0.10
+Address:	10.0.0.10#53
+
+nginx.default.svc.cluster.local	service = 10 50 0 web-1.ub.default.svc.cluster.local.
+nginx.default.svc.cluster.local	service = 10 50 0 web-0.ub.default.svc.cluster.local.
+```
+
+### Ordering
+
+While stable identites might decouple the physical infrastructure from units of compute, it's not sufficient to decouple system administration from application development. This is a problem the industry hasn't solved for many years, and we're probably not going to solve it now, but there are certain properties of an ordered set that we can leverage to significantly reduce the probability that someone wakes up to an alarming pager.
```

Comment:
Yeah, same thought.  Maybe reorganize to describe this as a hard problem, and we want to provide a simple and predictable framework for understanding how replacements are made.